src/workers/opportunity/storeCandidateApplicationScore.ts

```diff
@@ -0,0 +1,51 @@
+import { TypedWorker } from '../worker';
+import { TypeORMQueryFailedError } from '../../errors';
+import { ApplicationScored } from '@dailydotdev/schema';
+import { OpportunityMatch } from '../../entity/OpportunityMatch';
+import { applicationScoreSchema } from '../../common/schema/opportunities';
+
+export const storeCandidateApplicationScore: TypedWorker<'gondul.v1.candidate-application-scored'> =
+  {
+    subscription: 'api.store-candidate-application-score',
+    handler: async ({ data }, con, logger): Promise<void> => {
+      try {
+        const { userId, opportunityId, score, description } = data;
+        if (!userId || !opportunityId) {
+          logger.warn(
+            { data },
+            'Missing userId or opportunityId in candidate application score',
+          );
+          return;
+        }
+
+        const applicationRank = applicationScoreSchema.parse({
+          score,
+          description,
+        });
+
+        await con.transaction(async (manager) => {
+          await manager.getRepository(OpportunityMatch).upsert(
+            {
+              userId,
+              opportunityId,
+              applicationRank,
+            },
+            {
+              conflictPaths: ['userId', 'opportunityId'],
+              skipUpdateIfNoValuesChanged: true,
+            },
+          );
+        });
+      } catch (originalError) {
+        const err = originalError as TypeORMQueryFailedError;
+        if (err?.name === 'QueryFailedError') {
+          logger.error({ err, data }, 'could not update application score');
+
+          return;
+        }
```

Comment:
why would this not throw?

Reply:
It's just copy-paste of the other workers.
Better to log and catch no?

Reply:
which other worker? Usually if query fails we want to retry or inspect why it failed

Reply:
since sole purpose of worker is to update application score

Reply:
oh I see `storeCandidateOpportunityMatch`, honestly there I would also just throw

Reply:
The other one I did haha,  removed it from both now.